lib/rubocop/cop/style/redundant_arguments.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for redundant arguments to methods.
+      # Method names & arguments can be configured like:
+      #
+      # RedundantArguments:
+      #   join: ''
+      #   split: ' '
+      #   foo: 2
+      #
+      # @example
+      #   # bad
+      #
+      #   array.join('')
+      #   [1, 2, 3].join("")
+      #   string.split(" ")
+      #   "first\nsecond".split(" ")
+      #   A.foo(2)
+      #
+      #   # good
+      #   array.join
+      #   [1, 2, 3].join
+      #   string.split
+      #   "first second".split
+      #   A.foo
+      class RedundantArguments < Cop
+        MSG = 'Use `%<without_argument>s` instead of `%<with_argument>s`.'
+
+        def on_send(node)
+          arg = cop_config['RedundantArguments'][node.method_name.to_s]
```

Comment:
Might be a good idea to add some validation logic for the config, so we can present the users with more meaningful error messages if they mess something up.